### PR TITLE
[DOC] move example for surface API  plot_surface_image_and_maskers

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Closes none but relates to #4568

Changes proposed in this pull request:

- part of moving surface API to stable codebase
- move example from experimental
- add error and test in case users pass a SurfaceImage as a mesh to surface plotting functions (otherwise the error message makes debugging really hard)

### CODE REVIEW

**Comment by Remi-Gau** [comment]
re-reading the whole example I am wondering if it makes sense to keep it as is or if we should not better try to integrate it into the other examples

it mixes plotting, decoding, scikit learn pipelines... this feels a bit like a grab bag of things related to the new surface API

if we want to keep it as a stand alone then maybe as a release hightlights like scikit learn does it?

https://scikit-learn.org/stable/auto_examples/release_highlights/plot_release_highlights_1_5_0.html

**Comment by bthirion** [comment]
> re-reading the whole example I am wondering if it makes sense to keep it as is or if we should not better try to integrate it into the other examples
> 
> it mixes plotting, decoding, scikit learn pipelines... this feels a bit like a grab bag of things related to the new surface API
> 
> if we want to keep it as a stand alone then maybe as a release hightlights like scikit learn does it?
> 
> https://scikit-learn.org/stable/auto_examples/release_highlights/plot_release_highlights_1_5_0.html

I'm not sure who reads these release highlights.
But I understand it would be nicer to complement existing examples rather than add yet another one.

**Comment by Remi-Gau** [comment]
Will wait until all the other experimental examples have been moved to see if this one can broken down and integrated into other examples.

**Comment by Remi-Gau** [comment]
decided yesterday to move it into the 'advanced' example section and potentially refactor it after the next release

**Review by bthirion — APPROVED** [review]
LGTM, thx.

File: nilearn/plotting/tests/test_surf_plotting.py [hunk 1]
```diff
@@ -272,7 +272,7 @@ def test_check_surface_plotting_inputs_error_mash_and_data_none(fn):
         fn(None, None)
 
 
-def test_check_surface_plotting_inputs_errors():
+def test_check_surface_plotting_inputs_errors(surf_img):
     """Fail is mesh is none and data is not not SurfaceImage."""
```
Comment (man-shu) [inline nilearn/plotting/tests/test_surf_plotting.py:276]: minor typo
```suggestion
    """Fail if mesh is none and data is not not SurfaceImage."""
```

**Review by man-shu — APPROVED** [review]
LGTM!

**Pull request merged**